File: BACK_END_CAR/CAR_RENTAL_API/Entities/Rental.cs
using System.ComponentModel.DataAnnotations;

namespace CAR_RENTAL_API.Entities
{
    public class Rental
    {
        [Key]
        public Guid RentalId { get; set; }
        public Guid RentalRequestId { get; set; }
        public string RentalStatus { get; set; }
        public DateTime RentalDate { get; set; }
        public double OdometerStart { get; set; }
        public DateTime? ReturnDate { get; set; }  
        public double? OdometerEnd { get; set; }  
        public string? InspectionStatus { get; set; }
        public Payment Payment { get; set; }
        public Guid? PaymentId { get; set; }
    }

}

File: FRONT_END_CAR/src/app/Shared/models/rental.ts
import { Payment } from './payment';

export interface Rental {
  rentalId: string;
  rentalRequestId: string;
  rentalStatus: string;
  rentalDate: string;
  odometerStart: number;
  returnDate: string | null;
  odometerEnd: number | null;
  inspectionStatus: string | null;
  payment: Payment;
  paymentId: string;
}

export interface RentalTableRequestDTO {
  requestId: string;
  rentalDate: string;
  odometerStart: number;
  rentalPayment: number;
  advancePayment: number;
}
